torch/distributed/rpc.py

```diff
@@ -0,0 +1,104 @@
+from . import invoke_rpc
+from . import ProcessGroupAgent
+
+import array
+import torch
+
+_agent = None
+
+def _collect_worker_names(name, group):
+    from . import all_gather
+    from . import get_world_size
+
+    # collect name length
+    ws = get_world_size(group)
+    name_len = len(name)
+    len_input = torch.ones(1, dtype=torch.int64) * name_len
+    len_outputs = [torch.empty(1, dtype=torch.int64) for _ in range(ws)]
+    all_gather(len_outputs, len_input, group=group)
+
+    # collect name value
+    max_len = torch.stack(len_outputs).max().item()
+    name_input = torch.empty(max_len, dtype=torch.uint8)
+    name_input[:name_len] = torch.Tensor(list(name.encode('utf-8')))
+    name_outputs = [torch.empty(max_len, dtype=torch.uint8) for _ in range(ws)]
+    all_gather(name_outputs, name_input, group=group)
+
+    names = []
+    for i in range(ws):
+        name_tensor = name_outputs[i][:len_outputs[i]]
+        names.append(array.array('B', name_tensor.tolist()).tobytes().decode('utf-8'))
+
+    return names
+
+
+def init_rpc(name, backend='pg'):
+    r"""
+    Arguments:
+        name (str): name of this worker.
+        backend (str): type of RPC backend implementation.
+    """
+    global _agent
+    if backend == 'pg':
+        from . import is_initialized
+
+        assert is_initialized(), (
```

Comment:
Per the naming of this function, this is not an internal function. Then this error shouldn't be an assert, because assert failure implies we did something wrong. But a user can trivially trigger this by calling `init_rpc`. Make it a RuntimeError or something.